Tests fail on MacOS

## 🐛 Bug Report

Tests pass successfully on Linux, but fail to pass on MacOS. Currently 4 tests are failing:

```
       4 failed
         - tests/test_cli.py:56 ProcessIssuesTest.test_output_dir_is_called_out
         - tests/test_pre_commit.py:10 PreCommitTests.test_scan_is_executed_against_current_working_directory
         - tests/test_scan_local_repo.py:23 ScanLocalRepoTests.test_scan_exits_gracefully_when_target_is_not_git_repo
         - tests/test_scan_remote_repo.py:98 ScanRemoteRepoTests.test_subdir_of_work_dir_is_passed_to_clone_repo
```

An example error output is:

```
E           AssertionError: Expected call: clone_git_repo('git@github.com:godaddy/tartufo.git', PosixPath('/var/folders/p3/v5952rgx353dpjss7_fbldy00000gp/T/tmp2whea6u6/tartufo.git'))
E           Actual call: clone_git_repo('git@github.com:godaddy/tartufo.git', PosixPath('/private/var/folders/p3/v5952rgx353dpjss7_fbldy00000gp/T/tmp2whea6u6/tartufo.git'))
```
## To Reproduce

On MacOS, run the tests either with `pytest`, or `tox`.

## Expected Behavior

Tests should behave consistently across platforms!

## Code Example

N/A

## Environment

MacOS

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -71,7 +71,7 @@ class ProcessIssuesTest(unittest.TestCase):
             )
         self.assertEqual(
             result.output,
-            f"Results have been saved in {dirname}/foo/tartufo-scan-results-nownownow\n",
+            f"Results have been saved in {Path(dirname).resolve()}/foo/tartufo-scan-results-nownownow\n",
         )
 
     @mock.patch("tartufo.commands.scan_local_repo.GitRepoScanner")

--- a/tests/test_pre_commit.py
+++ b/tests/test_pre_commit.py
@@ -1,4 +1,5 @@
 import unittest
+from pathlib import Path
 from unittest import mock
 
 from click.testing import CliRunner
@@ -15,7 +16,9 @@ class PreCommitTests(unittest.TestCase):
         runner = CliRunner()
         with runner.isolated_filesystem() as tempdir:
             runner.invoke(cli.main, ["pre-commit"])
-        self.assertEqual(mock_scanner.call_args[0][1], tempdir)
+        self.assertEqual(
+            Path(mock_scanner.call_args[0][1]).resolve(), Path(tempdir).resolve()
+        )
 
     @mock.patch("tartufo.commands.pre_commit.GitPreCommitScanner")
     def test_scan_fails_on_scan_exception(self, mock_scanner: mock.MagicMock):

--- a/tests/test_scan_local_repo.py
+++ b/tests/test_scan_local_repo.py
@@ -1,4 +1,5 @@
 import unittest
+from pathlib import Path
 from unittest import mock
 
 from click.testing import CliRunner
@@ -25,5 +26,6 @@ class ScanLocalRepoTests(unittest.TestCase):
         with runner.isolated_filesystem() as dirname:
             result = runner.invoke(cli.main, ["scan-local-repo", "."])
             self.assertEqual(
-                result.output, f"{dirname} is not a valid git repository.\n"
+                result.output,
+                f"{Path(dirname).resolve()} is not a valid git repository.\n",
             )

--- a/tests/test_scan_remote_repo.py
+++ b/tests/test_scan_remote_repo.py
@@ -115,5 +115,6 @@ class ScanRemoteRepoTests(unittest.TestCase):
                 ],
             )
             mock_clone.assert_called_once_with(
-                "git@github.com:godaddy/tartufo.git", Path(dirname) / "tartufo.git"
+                "git@github.com:godaddy/tartufo.git",
+                Path(dirname).resolve() / "tartufo.git",
             )

